python/covid_web/covid_web/rest.py

```diff
@@ -79,24 +98,103 @@ def get_app(self):
         return self
 
     def get_available_predictions(self):
-        return jsonify(self.available_predictions)
+        return flask.jsonify(self.available_predictions)
 
-    def get_specific_prediction(self, date: str, country: str):
+    def _get_predictions_by_country(self, country: str):
+        result = [
+            prediction
+            for prediction in self.predictions
+            if prediction["short_name"] == country
+        ]
+
+        return result
+
+    def get_predictions_by_country(self, country: str):
+        result = self._get_predictions_by_country(country=country)
+
+        if len(result) == 0:
+            flask.abort(404)
+
+        return flask.jsonify(result)
+
+    def _get_predictions_by_name(self, prediction_name: str):
+        result = [
+            prediction
+            for prediction in self.predictions
+            if prediction["prediction_name"] == prediction_name
+        ]
+
+        return result
+
+    def get_predictions_by_name(self, date: str):
+        result = self._get_predictions_by_name(prediction_name=date)
+
+        if len(result) == 0:
+            flask.abort(404)
+
+        return flask.jsonify(result)
+
+    def _get_specific_prediction(self, date: str, country: str):
         for prediction in self.predictions:
-            if prediction["short_name"] == country and prediction["date_name"] == date:
-                return jsonify(prediction)
+            if prediction["short_name"] == country and prediction["prediction_name"] == date:
+                return prediction
 
-        abort(404)
         return ""
 
+    def get_specific_prediction(self, date: str, country: str):
+        result = self._get_specific_prediction(date=date, country=country)
+        if result == "":
+            flask.abort(404)
+
+        return flask.jsonify(result)
+
     def get_country_data(self, country: str):
         if country not in self.country_reports_active:
-            abort(404)
+            flask.abort(404)
 
-        return jsonify(self.country_reports_active[country])
+        return flask.jsonify(self.country_reports_active[country])
 
+    def generate_static_files(self, output_dir: Path) -> None:
+        Path(output_dir / f"data").mkdir(parents=True, exist_ok=True)
+        Path(output_dir / f"predictions/by_prediction").mkdir(parents=True, exist_ok=True)
+        Path(output_dir / f"predictions/by_country").mkdir(parents=True, exist_ok=True)
```

Comment:
```suggestion
        Path(output_dir / "data").mkdir(parents=True, exist_ok=True)
        Path(output_dir / "predictions" / "by_prediction").mkdir(parents=True, exist_ok=True)
        Path(output_dir / "predictions" / "by_country").mkdir(parents=True, exist_ok=True)
```

... and similarly at other places in this function.

Reply:
Applied.